statsmodels/tsa/vecm/vecm.py

```diff
@@ -0,0 +1,2146 @@
+
+        References
+        ----------
+        .. [1] Lutkepohl, H. 2005. *New Introduction to Multiple Time Series Analysis*. Springer.
+        """
+        param_names = []
+
+        if self.coint_rank == 0:
+            return None
+
+        # loading coefficients (alpha) # called "ec" in JMulTi, "ECT" in tsDyn,
+        param_names += [               # and "_ce" in Stata
+            self.load_coef_repr + "%d.%s" % (i+1, self.endog_names[j])
+            for j in range(self.neqs)
+            for i in range(self.coint_rank)
+        ]
+
+        return param_names
+
+    @property
+    def coint_param_names(self):
+        """
+
+        Returns
+        -------
+        param_names : list of str
+            Returns a list of parameter names for the cointegration matrix
+            as well as deterministic terms inside the cointegration relation
+            (if present in the model).
+
+        """
+        # 1. cointegration matrix/vector
+        param_names = []
+
+        param_names += [("beta.%d." + self.load_coef_repr + "%d") % (j+1, i+1)
+                        for i in range(self.coint_rank)
+                        for j in range(self.neqs)]
+
+        # 2. deterministic terms inside cointegration relation
+        if "ci" in self.deterministic:
+            param_names += ["const." + self.load_coef_repr + "%d" % (i+1)
+                            for i in range(self.coint_rank)]
+
+        if "li" in self.deterministic:
+            param_names += ["lin_trend." + self.load_coef_repr + "%d" % (i+1)
+                            for i in range(self.coint_rank)]
+
+        if self.exog_coint is not None:
+            param_names += ["exog_coint%d.%s" % (n+1, exog_no)
+                            for exog_no in range(1, self.exog_coint.shape[1]+1)
+                            for n in range(self.neqs)]
+
+        return param_names
+
+
+# -----------------------------------------------------------------------------
+# VECMResults class
+
+class VECMResults(object):
+    """Class holding estimation related results of a vector error correction
+    model (VECM).
+
+    Parameters
+    ----------
+    endog_tot : ndarray (neqs x nobs_tot)
+        Array of observations.
+    exog: ndarray (nobs_tot x neqs) or None
+        Deterministic terms outside the cointegration relation.
+    exog_coint: ndarray (nobs_tot x neqs) or None
+        Deterministic terms inside the cointegration relation.
+    k_ar : int
+        Lags in the VAR representation. This implies: Lags in the VEC
+        representation = k_ar - 1
+    coint_rank : int
+        Cointegration rank, equals the rank of the matrix :math:`\\Pi` and the
+        number of columns of :math:`\\alpha` and :math:`\\beta`.
+    alpha : ndarray (neqs x coint_rank)
+        Estimate for the parameter :math:`\\alpha` of a VECM.
+    beta : ndarray (neqs x coint_rank)
+        Estimate for the parameter :math:`\\beta` of a VECM.
+    gamma : ndarray (neqs x neqs*(k_ar-1))
```

Comment:
these three should be changed to something `params_xxx`

also we need informative names, and then a list of how they map to Luetkepohl's or Jorgensen's math notation